x/margin/keeper/msg_server.go

```diff
@@ -328,6 +326,31 @@ func (k Keeper) ForceCloseLong(ctx sdk.Context, msg *types.MsgForceClose, isAdmi
 	return &mtp, repayAmount, nil
 }
 
+func (k msgServer) AdminClose(goCtx context.Context, msg *types.MsgAdminClose) (*types.MsgAdminCloseResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	signer, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return nil, err
+	}
+	if !k.AdminKeeper().IsAdminAccount(ctx, admintypes.AdminType_MARGIN, signer) {
+		return nil, sdkerrors.Wrap(admintypes.ErrPermissionDenied, fmt.Sprintf("signer not authorised: %s", msg.Signer))
+	}
+
+	params := k.GetParams(ctx)
+	if msg.CloseAll {
```

Comment:
Do we need this boolean? If it's set to false then does the AdminClose just change the ForceCloseFundPercentage?

Reply:
Yeah , and then we can just use the existing force close to close individual MTP's
But I think this can be discussed. I do feel if Repay is not needed at all  , for any ForceClose , I can just change the ForceClose function to update this parameter as well , and that would make things much cleaner .